### DISCUSSION TITLE
Same PDF with a single difference which is configure to ignore is counted as different

### DISCUSSION THREAD

**Discussion by smyaseen** [red6/pdfcompare#103] [Q&A]
Hi, as the title says, I've to similar PDF just a single text changed, which I've also configured to ignore is still counted as different.

See the attached pic.

![image](https://user-images.githubusercontent.com/37702376/122747550-a0d95800-d2a4-11eb-8ae1-56fb21fc8a36.png)
![image](https://user-images.githubusercontent.com/37702376/122747694-cc5c4280-d2a4-11eb-9fba-4c4be51416a0.png)

This I've ignored. The PDF are same, I just copy pasted and changed a single text and also ignored it. I want this to be not counted as different.

![image](https://user-images.githubusercontent.com/37702376/122747821-f01f8880-d2a4-11eb-9064-8268cb6f55ab.png)

**Comment by finsterwalder** [comment]
What is in your ignore.conf file?
I think there is a small difference somewhere else, that you overlooked.
The Difference found message in the log shows the smallest rectangle, that includes all changes found.
It shows a rectangle with almost 1922 x 1163 pixes. Either you use a very high density or there is another difference somewhere.

**Reply by finsterwalder** [reply]
Sometimes there are very small differences, due to slight changes in font rendering.

**Reply by smyaseen** [reply]
There was no difference, as I said PDF were literally same just copy pasted. See here the pdf are same but still showing difference here

![image](https://user-images.githubusercontent.com/37702376/122753883-5065f880-d2ac-11eb-88f7-542842d779a4.png)


I found a solution. Although the PDFs were same, it was still counting as different to some text below with really slight greyed out. I had to mention a threshold with this

setAllowedDiffInPercent(0.1))
 
 Now it's working as expected. Thank you.

**Reply by finsterwalder** [reply]
If that works for you that's fine.
Otherwise it would need some research to actually find out, why the documents differ in the way shown.
Is it really an exact copy of the same file? Or do you use two different files from different sources?
Can you email me a copy of the PDF(s)?

**Reply by smyaseen** [reply]
Yes, mailing you. The PDFs were same, 

1) I have a.pdf 
2) I copied a.pdf , pasted and renamed it as b.pdf
3) Opened the b.pdf and changed a single word. I also mentioned that area in the ignore.conf, as you can see the area in the main question screenshot yellowed out.

4) Now, when I compare, it results as not equal, although the area is ignored. But shows difference in somewhere else now.
